scr/Parking-Zone.Data/Repositories/Repository.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Parking_Zone.Data.IRepositories;
+using Parking_Zone.Domain.Entities;
+
+namespace Parking_Zone.Data.Repositories;
+
+public class Repository<T> : IRepository<T> where T : Auditable
+{
+    protected readonly ApplicationDbContext _dbcontext;
+
+    protected readonly DbSet<T> _dbSet;
+
+    public Repository(ApplicationDbContext dbcontext)
+    {
+        _dbcontext = dbcontext;
+        _dbSet = _dbcontext.Set<T>();
+    }
+
+    public async Task<T> CreateAsync(T entity)
+    {
+        var entry = await _dbSet.AddAsync(entity);
+
+        await _dbcontext.SaveChangesAsync();
```

Comment:
in order to not call SaveChangesAsync(); in every mthod just add void Save(); in the IRepository

Reply:
Honestly, I could not understand how to apply this approach in Generic IRepository, i tried to add Task Save() method and implemented it into Repository, but without calling Save() method in every method, i could not save changes.